app/scripts/modules/core/src/presentation/forms/interface.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { FormikProps } from 'formik';
+
+export interface IFieldLayoutPropsWithoutInput extends IValidationProps {
+  label?: string | JSX.Element;
+  help?: JSX.Element;
+  required?: boolean;
+  actions?: JSX.Element;
+}
+
+export interface IFieldLayoutProps extends IFieldLayoutPropsWithoutInput {
+  input: JSX.Element;
+}
+
+export interface IValidationProps {
+  error?: string | JSX.Element;
+  warning?: string | JSX.Element;
+  preview?: string | JSX.Element;
+}
+
+export interface IFieldProps<T = string> extends IFieldLayoutPropsWithoutInput, IValidationProps {
```

Comment:
This is the base props that a `Field` component receives. 

This combines: 

- Validation Props
  - `error`
  - `warning`
  - `preview`
- Layout Props (except `input`)
  - `label`
  - `help`
  - `required`
  - `actions`
- state management Props
  - `value`
  - `onChange`
- The layout component to use, if not the default one
  - `FieldLayout`